exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -237,7 +250,11 @@ func (prwe *prwExporter) execute(ctx context.Context, writeReq *prompb.WriteRequ
 	req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
 	req.Header.Set("User-Agent", prwe.userAgentHeader)
 
-	resp, err := prwe.client.Do(req)
+	retryReq, err := retryablehttp.FromRequest(req)
```

Comment:
Hi why we should use retryable http here?

I think the main issue here is the error is treated as permanent error: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusremotewriteexporter/exporter.go#L206

And as my understanding, shouldn't retry logic is handled in here?

https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/queued_retry.go#L390

Reply:
This component cannot rely on the queue_retry because it internally submit requests of up to a certain size [1] in parallel [2] to the backend. We are controlling the parallelism internally because the we want to guarantee that samples of a same time series are always ingested in chronological order [3].


[1] https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusremotewriteexporter/helper.go#L14
[2] https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusremotewriteexporter/exporter.go#L193
[3] https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusremotewriteexporter/factory.go#L43

As a side effect, we cannot rely on the queue retry mechanism of the exporterhelper.

Reply:
I see. Thanks for the detailed explanation.